### DISCUSSION TITLE
How does presto read local files in parquet format

### DISCUSSION THREAD

**Discussion by jiangjiguang** [prestodb/presto#18182] [Q&A] [answered]
presto  has the "Local File Connector" , but maybe it only reads http_request_log. 

I do not know how to read l local file  in parquet format

**Comment by majetideepak** [comment, marked as answer]
@jiangjiguang there is an undocumented hive connector feature where you can set the following config and use a local directory as the catalog.
```
hive.metastore=file
hive.metastore.catalog.dir=file:/mnt/hive_data/
```
Now, you have to create a table say `customer` using the parquet file schema. 
You will see a folder with this table name `/mnt/hive_data/customer`
Then move your parquet file to this folder. You should be able to query the Parquet file.

We should also have a formal way in Presto to read local Parquet/ORC files. I will look into formalizing the `hive.metastore=file`.
